Common/OvermorrowGlobalItem.cs

```diff
@@ -1,11 +1,284 @@
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using OvermorrowMod.Common.Particles;
+using OvermorrowMod.Content.Items.Consumable;
+using OvermorrowMod.Core;
+using System;
 using Terraria;
+using Terraria.GameContent;
 using Terraria.ID;
 using Terraria.ModLoader;
 
 namespace OvermorrowMod.Common
 {
     public class OvermorrowGlobalItem : GlobalItem
     {
+        public override bool InstancePerEntity => true;
+
+        public int warningDelay = 0;
+        public float reforgeAnimation = 0;
+        public int resetDelay = 0;
+
+        private bool drawInvalid = false;
+        private TParticleSystem sys = new();
+        private Color reforgeColor = Color.White;
+
+        private float DPSCalculation(Item item)
+        {
+            return (float)(item.damage * 60f / item.useTime) * (1 + item.crit / 100f);
+        }
+
+        public override bool ConsumeItem(Item item, Player player)
+        {
+            if (Main.mouseItem.type == ModContent.ItemType<ReforgeStone>())
+            {
+                return false;
+            }
+
+            return base.ConsumeItem(item, player);
+        }
+
+        public override bool CanRightClick(Item item)
+        {
+            if (Main.mouseItem.ModItem is ReforgeStone)
```

Comment:
Is this necessary? Can't you do the `CanRightClick` stuff in `ReforgeStone` directly? It seems kinda silly to have a global item that only affects a single mod item from the current mod, not what they are for.

Reply:
The `CanRightClick` is not for the `ReforgeStone` itself, but for the item that the stone is attempting to apply to. 

For instance, if I have a sword and I am right clicking it while holding a stone then it is the sword that is processing that input and then uses its parameter `Item` instance to modify itself. I couldn't figure out how to grab that direct sword instance otherwise.

Reply:
Ah, makes sense.